bundles/org.openhab.binding.solarman/src/main/java/org/openhab/binding/solarman/internal/modbus/SolarmanLoggerConnection.java

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.solarman.internal.modbus;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.SocketAddress;
+import java.net.SocketTimeoutException;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Catalin Sanda - Initial contribution
+ */
+@NonNullByDefault
+public class SolarmanLoggerConnection implements AutoCloseable {
+    private final Logger logger = LoggerFactory.getLogger(SolarmanLoggerConnection.class);
+    private boolean allowLogging;
+    @Nullable
+    private Socket socket;
+
+    public SolarmanLoggerConnection(String hostName, int port, boolean allowLogging) {
+        SocketAddress sockaddr = new InetSocketAddress(hostName, port);
+        Socket localSocket = connectSocket(sockaddr);
+
+        if (localSocket == null) {
+            if (allowLogging) {
+                logger.error("Error creating socket");
+            }
+        } else {
+            socket = localSocket;
+        }
+
+        this.allowLogging = allowLogging;
+    }
+
+    public byte[] sendRequest(byte[] reqFrame) {
+        // Will not be used by multiple threads, so not bothering making it thread safe for now
+        Socket localSocket = socket;
+
+        if (localSocket == null) {
+            if (allowLogging) {
+                logger.error("Socket is null, not reading data this time");
```

Comment:
```suggestion
                logger.debug("Socket is null, not reading data this time");
```
Use custom exceptions if you want to route this to a `Thing` state and inform the user. This looks liek a transient issue and i would write it to dbeug that is all.